packages/flutter/lib/src/material/list_tile.dart

```diff
@@ -869,9 +869,13 @@ class ListTile extends StatelessWidget {
           defaults.textColor,
           theme.disabledColor,
         );
+
+    final IconButtonThemeData currentIconTheme = IconButtonTheme.of(context);
     final IconThemeData iconThemeData = IconThemeData(color: effectiveIconColor);
     final IconButtonThemeData iconButtonThemeData = IconButtonThemeData(
-      style: IconButton.styleFrom(foregroundColor: effectiveIconColor),
+      style: currentIconTheme.style?.copyWith(
+        foregroundColor: MaterialStateProperty.all(effectiveIconColor),
+      ),
```

Comment:
Is `currentIconTheme.style` guaranteed to be non-null here? Otherwise a fallback would be necessary to set the `iconColor`:

```suggestion
      style: currentIconTheme.style?.copyWith(
        foregroundColor: MaterialStateProperty.all(effectiveIconColor),
      ) ?? IconButton.styleFrom(foregroundColor: effectiveIconColor),
```

Reply:
currentIconTheme is created right above these lines. But I added the fallback :)